### PR TITLE
android-tools: 31.0.3p1 -> 33.0.3

### PR DESCRIPTION
**Pull request by SebTM**
###### Description of changes
Updated to latest release (see #196144 for discussions)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SebTM** [comment]
Is there a reason we don't build this from source? (I didn't got it working on short try)

**Comment by winterqt** [comment]
We do build these from source. (What makes you think otherwise?)

**Comment by SebTM** [comment]
> We do build these from source. (What makes you think otherwise?)

Oh sorry then it's my fault because we have to fetch the release archives and switching to fetchFromGitHub e.g. seems to miss steps/files so I guessed it has something prebuild we can't do

**Comment by winterqt** [comment]
Yeah, the source tarballs contain everything already patched for us. Per [the README](https://github.com/nmeum/android-tools#installation):
> Source tarballs containing an already patched version of all vendored dependencies are available on the [GitHub Release Page](https://github.com/nmeum/android-tools/releases).
>
> These tarballs should be used for packaging and general installation.

**Review by winterqt — CHANGES_REQUESTED** [review]
Per https://github.com/NixOS/nixpkgs/pull/196144#discussion_r999897390, can you remove `fmt_7` as a dependency (since there's no option to disable it, and we may be better off to continue using Google's version)?

**Comment by SebTM** [comment]
Thanks for the explanation :) Adjusted according feedback 👍🏻

**Review by yu-re-ka — APPROVED** [review]

File: pkgs/tools/misc/android-tools/default.nix [hunk 1]
```diff
@@ -9,22 +9,14 @@ in
 
 stdenv.mkDerivation rec {
   pname = "android-tools";
-  version = "31.0.3p1";
+  version = "33.0.3";
 
   src = fetchurl {
     url = "https://github.com/nmeum/android-tools/releases/download/${version}/android-tools-${version}.tar.xz";
-    sha256 = "1f2svy381r798hjinrc2xiwz13gkkqxfill343zvv8jqkn8rzxhf";
+    sha256 = "jOF02reB1d69Ke0PllciMfd3vuGbjvPBZ+M9PqdnC8U=";
```
Comment (Atemu) [inline pkgs/tools/misc/android-tools/default.nix:16]:
```suggestion
    hash = "sha256-jOF02reB1d69Ke0PllciMfd3vuGbjvPBZ+M9PqdnC8U=";
```

**Review by Atemu — APPROVED** [review]

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 199212` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>agi</li>
    <li>android-tools</li>
  </ul>
</details>

**Review by Atemu — APPROVED** [review]

**Comment by Atemu** [comment]
@winterqt could you remove your stale review?

**Comment by winterqt** [comment]
> @winterqt could you remove your stale review?

Why? I'm going to merge it anyways :)

I'm just waiting on the final OfBorg build to sanity check.

**Comment by Atemu** [comment]
@winterqt it's just for clarity's sake.

**Comment by winterqt** [comment]
> @winterqt it's just for clarity's sake.

Makes sense, apologies.

**Review by winterqt — APPROVED** [review]
Thanks!

**Pull request merged**